resolver/blocking_resolver.go

```diff
@@ -95,7 +95,7 @@ type BlockingResolver struct {
 // NewBlockingResolver returns a new configured instance of the resolver
 func NewBlockingResolver(
 	cfg config.BlockingConfig, redis *redis.Client, bootstrap *Bootstrap,
-) (r *BlockingResolver, err error) {
+) (r ChainedResolver, err error) {
```

Comment:
I'm fine with always returning the struct pointer.

I've changed all resolver to return the interface type, because I was confused what my new resolver should return, as some of the existing resolvers were returning a struct pointer and others the interface.
I didn't know about the "Accept interfaces, return structs" pattern until now, thanks :slightly_smiling_face: 

If you want I can refactor all resolvers to always return its struct pointer.

Reply:
Yes, please change the return type to struct and remove casting in tests :+1: